### PR TITLE
Config file reader uses UTF_8 charset while reading the file #3164

### PR DESCRIPTION
**Pull request by maheshp**
**WIP - Do not merge**
* With this change, both of ConfigReader and ConfigWriter would use the
  same charest

### CODE REVIEW

File: server/src/com/thoughtworks/go/config/GoConfigFileReader.java [hunk 1]
```diff
@@ -21,16 +21,19 @@
...
     private final SystemEnvironment systemEnvironment;
+    private final Charset UTF_8 = StandardCharsets.UTF_8;
 
     public GoConfigFileReader(SystemEnvironment systemEnvironment) {
         this.systemEnvironment = systemEnvironment;
     }
 
     public String configXml() throws IOException {
-        return FileUtils.readFileToString(fileLocation());
+        return FileUtils.readFileToString(fileLocation(), UTF_8);
```
Comment (ketan) [inline server/src/com/thoughtworks/go/config/GoConfigFileReader.java]: Simply use static import instead of a constant that points to another one?

**Comment by jyotisingh** [comment]
There are a few other places too which rely on the default-charset as far as the config xml is concerned. 

* During [checkin](https://github.com/gocd/gocd/blob/4ef23ba6cba377a4896562c60d8a31f680955f25/config/config-server/src/com/thoughtworks/go/service/ConfigRepository.java#L110) of the config xml to the config.git repo, default-charset is assumed there. 
* If the config file was found to be invalid during a server startup, it reverts itself from the last known good revision from config.git. Check [this](https://github.com/gocd/gocd/blob/4ef23ba6cba377a4896562c60d8a31f680955f25/config/config-server/src/com/thoughtworks/go/config/GoConfigMigration.java#L148) and [this](https://github.com/gocd/gocd/blob/4ef23ba6cba377a4896562c60d8a31f680955f25/config/config-server/src/com/thoughtworks/go/config/GoConfigMigration.java#L160). 
* Config upgrade (xsl migration) when a server upgrades - check [this](https://github.com/gocd/gocd/blob/4ef23ba6cba377a4896562c60d8a31f680955f25/config/config-server/src/com/thoughtworks/go/config/GoConfigMigration.java#L123)

**Comment by maheshp** [comment]
@lijie7 any changes to the Materials through the UI are written to config xml by explicitly using ```UTF-8``` encoding but the read from the config xml relies on the Default encoding of the OS.

So our assumption here is that since a single encoding is not used for both reads and writes there could be distortion as in your case.

While we make the necessary code changes to fix this issue, will you be able to change the locale of your OS to use ```UTF-8``` encoding to see if this issue gets resolved.

**Comment by maheshp** [comment]
This needs some more thought and testing, closing for now.

**Pull request closed without merging**